platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -146,6 +146,9 @@ - (void)startWithQueue:(dispatch_queue_t)queue completionHandler:(MGLMapSnapshot
     });
 
     dispatch_async(queue, ^{
+        if (!weakSelf) {
+            return;
```

Comment:
This introduces a race condition in which `weakSelf` could potentially be nilled out between here and the next line, where `weakSelf` (now `nil`) is assigned to `strongSelf`.